spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/resource/LRUCleaningResourceLoader.java

```diff
@@ -0,0 +1,122 @@
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.Assert;
+import org.springframework.util.FileSystemUtils;
+
+/**
+ * A wrapper around a {@link org.springframework.core.io.ResourceLoader} that deletes returned Resources (assumed to
+ * be on the file system) once disk space is getting low. Least Recently Used entries are removed first.
+ *
+ * <p>This wrapper is typically meant to be used to clean Maven {@literal .m2/repository} entries, but also works
+ * with other files. For the former case, if entries are under the configured {@link #repositoryCache} path (typically
+ * the {@literal .m2/repository} path), then the whole parent directory of the resource is removed. Otherwise, the sole
+ * resource file is deleted.</p>
+ *
+ * @author Eric Bottard
+ */
+// NOTE: extends DelegatingResourceLoader as a
+// work around https://github.com/spring-cloud/spring-cloud-dataflow/issues/1064 for now
+/*default*/ class LRUCleaningResourceLoader /*implements ResourceLoader*/ extends DelegatingResourceLoader {
+
+	private static final Logger logger = LoggerFactory.getLogger(LRUCleaningResourceLoader.class);
+
+	private final File repositoryCache;
+
+	private final ResourceLoader delegate;
+
+	private Map<File, Void> lruCache = this.new LRUCache();
+
+	private final float targetFreeSpaceRatio;
+
+	public LRUCleaningResourceLoader(ResourceLoader delegate, float targetFreeSpaceRatio, File repositoryCache) {
+		Assert.notNull(delegate, "delegate cannot be null");
+		Assert.isTrue(0 < targetFreeSpaceRatio && targetFreeSpaceRatio < 1, "targetFreeSpaceRatio should be between 0 and 1");
+		this.targetFreeSpaceRatio = targetFreeSpaceRatio;
+		this.delegate = delegate;
+		this.repositoryCache = repositoryCache;
+	}
+
+	@Override
+	public Resource getResource(String location) {
+		Resource resource = delegate.getResource(location);
+		try {
+			File file = resource.getFile();
+			synchronized (lruCache) {
+				lruCache.put(file, null);
+			}
+			return resource;
+		}
+		catch (IOException e) {
+			throw new RuntimeException(getClass().getSimpleName()
+				+ " is meant to work with File resolvable Resources. Exception trying to resolve " + location, e);
+		}
+	}
+
+	@Override
+	public ClassLoader getClassLoader() {
+		return delegate.getClassLoader();
+	}
+
+	private class LRUCache extends LinkedHashMap<File, Void> {
+
+		LRUCache() {
+			super(5, .75f, true);
+		}
+
+		@Override
+		protected boolean removeEldestEntry(Map.Entry<File, Void> eldest) {
+			for (Iterator<File> it = keySet().iterator(); it.hasNext(); ) {
+				File file = it.next();
+				logger.debug("Looking at {}, {} / {} = {}% free space", file, file.getFreeSpace(), file.getTotalSpace(), 100f * file.getFreeSpace() / file.getTotalSpace());
```

Comment:
To be somewhat proactive wrt to this feature 'in the wild' shall we make it info level?